lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -864,12 +868,12 @@ public function expandCriteriaParameters(Criteria $criteria)
         list($params, $types) = $valueVisitor->getParamsAndTypes();
 
         foreach ($params as $param) {
-            $sqlParams[] = PersisterHelper::getIdentifierValues($param, $this->em);
+            $sqlParams = array_merge($sqlParams, $this->getValues($param));
         }
 
         foreach ($types as $type) {
-            list($field, $value) = $type;
-            $sqlTypes[]          = $this->getType($field, $value, $this->class);
+            list ($field, $value) = $type;
+            $sqlTypes = array_merge($sqlTypes, $this->getTypes($field, $value, $this->class));
```

Comment:
Types seem to be incorrectly bound here, as the value conversion tests (for association columns) fail.